packages/vscode-extension/src/debugging/DebugSession.ts

```diff
@@ -89,16 +91,21 @@ export class DebugSession implements Disposable {
     return this.currentWsTarget;
   }
 
+  private startParentSessionPromise: Promise<vscode.DebugSession> | undefined;
```

Comment:
wonder why we add this when handling parent session but not for js session, seems like the problem with concurrent starting can happen for both of those sessions

Reply:
Mostly because issues when launching multiple JS sessions _kind of_ resolve themselves, since the debugged application would automatically disconnect the extra sessions.
There are potential issues with race conditions (when we save the session that disconnected instead of the one that succeeded) but they didn't come up in testing.

That said, there isn't a good reason (other than making the code harder to follow) to _not_ do it, so I changed it to use the cancel token as well when I changed the approach according to your other suggestion.